reference_data/management/commands/update_mgi.py

```diff
@@ -22,20 +22,21 @@ def __init__(self, **kwargs):
 
     @staticmethod
     def get_file_header(f):
-        return ['gene_symbol', 'entrez_gene_id', 'homologene_id', 'marker_id', 'phenotype_ids']
+        return ['gene_symbol', 'entrez_gene_id', 'mouse_gene_symbol', 'marker_id', 'phenotype_ids']
 
     @staticmethod
     def parse_record(record):
         yield {k: v.strip() for k, v in record.items() if k in ['gene_symbol', 'marker_id', 'entrez_gene_id']}
 
     def get_gene_for_record(self, record):
-        gene = self.entrez_id_to_gene.get(record.pop('entrez_gene_id'))
-        if gene:
-            del record['gene_symbol']
-            return gene
-
-        return super(MGIReferenceDataHandler, self).get_gene_for_record(record)
-
+        entrez_gene = self.entrez_id_to_gene.get(record.pop('entrez_gene_id'))
+
+        try:
+            return super(MGIReferenceDataHandler, self).get_gene_for_record(record)
+        except ValueError as e:
+            if entrez_gene:
+                return entrez_gene
+            raise e
```

Comment:
When returns `entrez_gene`, the `record['gene_symbol']` won't be deleted. It is different from the code before. Is that what you want?

Reply:
it will be because `super(MGIReferenceDataHandler, self).get_gene_for_record(record)` does that

Reply:
okay